src/pages/instances/InstanceSearchFilter.tsx

```diff
@@ -0,0 +1,92 @@
+import React, { FC } from "react";
+import { LxdInstance, LxdInstanceStatus } from "types/instance";
+import {
+  InstanceFilters,
+  instanceStatuses,
+  instanceTypes,
+} from "util/instanceFilter";
+import { SearchAndFilter } from "@canonical/react-components";
+import {
+  SearchAndFilterData,
+  SearchAndFilterChip,
+} from "@canonical/react-components/dist/components/SearchAndFilter/types";
+import { InstanceListType } from "util/instanceOptions";
+import isEqual from "lodash/isEqual";
+
+interface Props {
+  instances: LxdInstance[];
+  filters: InstanceFilters;
+  setFilters: (newFilters: InstanceFilters) => void;
+}
+
+const InstanceSearchFilter: FC<Props> = ({
+  instances,
+  filters,
+  setFilters,
+}) => {
+  const profileSet = [
+    ...new Set(instances.flatMap((instance) => instance.profiles)),
+  ];
+
+  const searchAndFilterData: SearchAndFilterData[] = [
+    {
+      id: 1,
+      heading: "Status",
+      chips: instanceStatuses.map((status) => {
+        return { lead: "status", value: status };
+      }),
+    },
+    {
+      id: 2,
+      heading: "Instance type",
+      chips: instanceTypes.map((type) => {
+        return { lead: "type", value: type.uiName };
+      }),
+    },
+    {
+      id: 3,
+      heading: "Profile",
+      chips: profileSet.map((profile) => {
+        return { lead: "profile", value: profile };
+      }),
+    },
+  ];
+
+  const onSearchDataChange = (searchData: SearchAndFilterChip[]) => {
+    const newQueries = searchData
+      .filter((chip) => chip.quoteValue)
+      .map((chip) => chip.value);
+    const newStatuses = searchData
+      .filter((chip) => chip.lead === "status")
+      .map((chip) => chip.value as LxdInstanceStatus);
+    const newTypes = searchData
+      .filter((chip) => chip.lead === "type")
+      .map((chip) => {
+        return {
+          name: chip.value === "VM" ? "virtual-machine" : "container",
+          uiName: chip.value,
+        } as InstanceListType;
+      });
+    const newProfileQueries = searchData
+      .filter((chip) => chip.lead === "profile")
+      .map((chip) => chip.value);
+    const newFilters = {
+      queries: newQueries,
+      statuses: newStatuses,
+      types: newTypes,
+      profileQueries: newProfileQueries,
+    };
+    if (!isEqual(filters, newFilters)) setFilters(newFilters);
```

Comment:
`onSearchDataChange` only gets called when some filter changed, no? I think we can drop the `isEqual` check here.

Reply:
Try it yourself, no? :)

Reply:
Well the prop comment says `A function that is called when the search data changes.` so I am pretty sure. :)

Reply:
I'm also sure since I tried it and got the "maximum update depth exceeded" error. That's why I said to try it.

Reply:
Right, because we feed in `filter` and `setFilters` the whole component re-renders. We don't need filters and can just remove it.
Then `setFilter` still changes every time we call it and causes a re-render. But we can avoid that with `export default React.memo(InstanceSearchFilter);`. Then we can simplify the interface to this component (by removing the filters prop) and get rid of the `isEqual` check.